### PR TITLE
Issue 189 - Lineage read API

### LINKED ISSUE #189: [TV3-DATA-001] Lineage complet run → set → décision → trade
## Parent et références

- Roadmap parent : #173
- Documentation de pilotage : #187 — `docs/roadmap/tradingv3-roadmap-v2.md`
- PR historique à reprendre/superséder proprement : #181
- Dépendance de recette : #188

## Contexte

Le nouvel orchestrateur possède un `run_id` et traite plusieurs sets dans un même run. Symfony produit ensuite des décisions, ordres, positions et événements de cycle de vie.

La corrélation partielle par symbole ou par `run_id` ne suffit pas lorsque :

- un run contient plusieurs sets ;
- plusieurs profils ciblent le même symbole ;
- plusieurs exchanges ou market types sont exécutés ;
- un set est rejoué ;
- un ordre est partiellement rempli ;
- une position génère plusieurs événements ;
- un traitement passe par Messenger ou un workflow asynchrone.

Sans lineage complet, il devient impossible de répondre de manière fiable à :

```text
Quel dashboard et quel set ont produit ce trade ?
Quelle config/profil/exchange étaient utilisés ?
Quel run a produit cette décision ?
Un replay a-t-il créé un doublon ?
Quel résultat PnL attribuer à chaque set ?
```

## Objectif

Propager et persister un contexte d’orchestration immuable depuis l’appel Python jusqu’aux décisions et événements de trade.

Le lineage cible est :

```text
dashboard_id
  -> orchestration_run_id
      -> orchestration_set_id
          -> correlation_run_id
              -> trading_decision_id
                  -> order_intent_id / order_id
                      -> position_id / trade_id
                          -> lifecycle events
```

Cette issue cadre le lineage. Elle ne doit pas être déclarée terminée par une simple propagation de `X-Run-Id`.

## Invariants

- Les identifiants reçus ne sont jamais tronqués.
- Le `run_id` de l’orchestrateur reste identique de bout en bout.
- Le `set_id` doit être distinguable dans un run multi-sets.
- Le profil, l’exchange, le market type et le symbole sont persistés avec le contexte.
- Un replay est distinguable du run initial sans casser la corrélation.
- Une absence de contexte legacy reste supportée, mais elle est explicitement identifiée comme legacy.
- Un conflit entre headers et payload est refusé ou audité selon une règle documentée ; aucun choix silencieux.
- Aucune donnée sensible n’est propagée dans les headers ou les logs.

## Contrat d’entrée HTTP proposé

L’orchestrateur transmet au minimum :

```text
X-Run-Id
X-Run-Correlation-Id
X-Orchestration-Set-Id
X-Orchestration-Dashboard-Id
```

Le contexte métier provenant du set matérialisé doit également contenir :

```text
mtf_profile
exchange
market_type
symbol ou liste de symboles
execution_mode / dry_run
config_effective_version ou hash, lorsque disponible
```

#### `X-Run-Id`

Identifiant canonique du run parent créé par l’orchestrateur.

- il est copié sans transformation ;
- il ne doit pas être remplacé par un UUID Symfony ;
- il reste identique pour tous les sets du même run.

#### `X-Orchestration-Set-Id`

Identifiant canonique du set persistant.

- obligatoire pour le chemin orchestrateur ;
- absent uniquement sur les chemins legacy explicitement supportés ;
- permet les agrégats par set.

#### `X-Orchestration-Dashboard-Id`

Identifiant du dashboard propriétaire du set.

- permet les agrégats par dashboard ;
- doit être vérifié contre la relation set/dashboard connue de l’orchestrateur.

#### `X-Run-Correlation-Id`

Identifiant déterministe du traitement d’un set dans un run.

Contrat recommandé :

```text
correlation_run_id = deterministic(run_id, set_id)
```

Contraintes :

- aucun découpage ou truncation ;
- format documenté et stable ;
- même valeur lors d’un retry technique du même run_set ;
- un replay fonctionnel doit conserver une référence vers l’origine et disposer d’un identifiant de replay/tentative séparé ;
- l’idempotence ne doit pas dépendre d’une chaîne ambiguë.

## Propagation Symfony attendue

Le contexte doit traverser explicitement :

1. contrôleur/endpoint `/api/mtf/run` ;
2. DTO de requête ;
3. runner applicatif ;
4. commande/message Messenger si utilisé ;
5. décision TradingCore / TradingDecision ;
6. création d’OrderIntent ;
7. soumission d’ordre ;
8. ouverture de position ;
9. événements de cycle de vie ;
10. clôture et calcul d’outcome.

Le contexte ne doit pas être reconstruit plus tard par recherche sur le symbole.

## Modèle de données attendu

Prévoir des champs dédiés, indexables et nullable pour la compatibilité legacy :

```text
orchestration_run_id
orchestration_set_id
orchestration_dashboard_id
correlation_run_id
mtf_profile
exchange
market_type
symbol
trading_decision_id
order_intent_id
exchange_order_id
position_id
trade_id
origin
replay_of_run_id ou replay_of_correlation_id
attempt_number
```

### Règles

- les identifiants structurants ne doivent pas vivre uniquement dans un JSON `extra` ;
- les événements peuvent conserver un snapshot JSON complémentaire ;
- les colonnes utilisées pour les jointures doivent être indexées ;
- les migrations sont additives et compatibles avec les données existantes ;
- les champs legacy manquants restent `NULL`, avec `origin=legacy` lorsque possible.

## Cohérence headers / payload

Ajouter une validation explicite :

- `set_id` header différent du set matérialisé → refus ;
- `dashboard_id` incohérent avec le set → refus ;
- `mtf_profile`, `exchange` ou `market_type` divergents → refus ou erreur structurée ;
- symbole hors périmètre du set → refus ;
- `dry_run` demandé mais payload live → fail-closed ;
- contexte live sans runtime-check → fail-closed.

L’erreur doit inclure un code stable, par exemple :

```text
ORCHESTRATION_CONTEXT_MISMATCH
ORCHESTRATION_SET_NOT_FOUND
ORCHESTRATION_DASHBOARD_MISMATCH
ORCHESTRATION_PROFILE_MISMATCH
ORCHESTRATION_EXCHANGE_MISMATCH
ORCHESTRATION_MARKET_TYPE_MISMATCH
```

## Compatibilité legacy

Les entrées CLI, cron ou HTTP historiques sans headers doivent continuer à fonctionner pendant la migration.

Comportement attendu :

- créer un contexte local clairement marqué `origin=legacy` ;
- ne jamais prétendre qu’il provient d’un set d’orchestration ;
- ne pas remplir artificiellement `set_id` ou `dashboard_id` ;
- conserver la compatibilité des signatures publiques autant que possible ;
- ajouter des deprecation logs exploitables ;
- documenter la date/condition de retrait du fallback.

## Idempotence et replay

Le lineage doit soutenir l’idempotence sans la remplacer.

À vérifier :

- une même combinaison idempotente ne crée pas deux décisions ;
- un retry technique réutilise le même contexte ;
- un replay explicite crée une nouvelle tentative liée à l’origine ;
- la clé d’idempotence métier `{exchange}:{market_type}:{symbol}:{timeframe}:{candle_open_ts}` reste compatible ;
- `run_id` ou `set_id` seuls ne sont pas considérés comme clé métier suffisante ;
- un même symbole traité par deux profils reste identifiable sans autoriser une exposition incompatible.

## Audit et logs

Chaque log structuré significatif doit pouvoir inclure, lorsqu’ils existent :

```text
orchestration_run_id
orchestration_set_id
orchestration_dashboard_id
correlation_run_id
mtf_profile
exchange
market_type
symbol
trading_decision_id
trade_id / position_id
origin
```

Contraintes :

- pas de secret ;
- pas de payload complet non redacted ;
- pas de génération d’un nouvel identifiant à chaque couche ;
- même nom de champ dans tous les services lorsque possible.

## API de lecture attendue

Prévoir des lectures permettant :

- outcome par `run_id` ;
- outcome par `set_id` ;
- outcomes d’un dashboard ;
- décisions et trades d’un `correlation_run_id` ;
- distinction des retries et replays ;
- filtres par profil, exchange, market type et symbole.

Le contrat API doit distinguer :

```text
run sans trade
run avec décisions mais sans ordre
run avec ordre non fillé
run avec position ouverte
run avec trade clôturé
run avec erreur
```

### Unitaires

- parsing des headers ;
- validation des formats ;
- génération déterministe de `correlation_run_id` ;
- absence de truncation ;
- détection de mismatch ;
- fallback legacy ;
- propagation dans les DTO/messages/events.

### Intégration

- orchestrateur → endpoint Symfony ;
- endpoint → décision ;
- décision → OrderIntent ;
- OrderIntent → lifecycle events ;
- clôture → outcome ;
- lecture par run et par set.

### Cas limites

- deux sets du même run sur le même symbole ;
- deux profils différents ;
- deux exchanges différents ;
- retry réseau ;
- replay explicite ;
- ordre partiellement rempli ;
- position clôturée en plusieurs fills ;
- ancien événement sans lineage ;
- headers valides mais payload contradictoire.

## Livrables

- [ ] Contrat de lineage documenté dans `docs/`.
- [ ] DTO/Value Object dédié au contexte d’orchestration.
- [ ] Propagation complète jusqu’aux lifecycle events.
- [ ] Migrations additives et index.
- [ ] Validation fail-closed des incohérences.
- [ ] Compatibilité legacy documentée et testée.
- [ ] API de lecture par run et set.
- [ ] Tests unitaires et d’intégration.
- [ ] Mise à jour du cockpit ou contrat prêt pour le cockpit.

## Critères d’acceptation

- [ ] Un trade clôturé est rattachable sans heuristique à son `run_id`, `set_id` et `dashboard_id`.
- [ ] Le profil, l’exchange et le market type sont disponibles sur la décision et l’outcome.
- [ ] Aucun identifiant reçu n’est tronqué ou remplacé.
- [ ] Deux sets du même run restent distinguables.
- [ ] Un retry technique ne crée pas un nouveau lineage.
- [ ] Un replay est explicitement lié au run d’origine.
- [ ] Les incohérences header/payload sont refusées avec un code stable.
- [ ] Les chemins CLI/legacy continuent à fonctionner avec `origin=legacy`.
- [ ] Les tests prouvent l’absence de reconstruction par symbole.
- [ ] La PR ne modifie aucune règle de stratégie, EntryZone, Risk/Leverage ou SL/TP.

## Découpage PR atomique recommandé

1. **PR 1 — Contrat et Value Object de contexte** : aucun changement de persistance.
2. **PR 2 — Headers Python + parsing Symfony** : propagation HTTP uniquement.
3. **PR 3 — Propagation applicative et Messenger**.
4. **PR 4 — Colonnes/migrations/index**.
5. **PR 5 — Lifecycle events et outcome par set**.
6. **PR 6 — Compatibilité legacy et deprecation logs**.
7. **PR 7 — API de lecture / intégration cockpit**.

Chaque PR doit rester reviewable et ne pas mélanger la fiabilisation de `position_trade_analysis`, traitée séparément dans `TV3-DATA-002`.

## Hors périmètre explicite

- stratégie MTF ;
- règles d’entrée ;
- EntryZone ;
- position sizing ;
- levier ;
- SL/TP ;
- activation live ;
- refonte YAML ;
- refonte Front Ops ;
- modèle de PnL net ;
- correction de la vue `position_trade_analysis`.

### PR DESCRIPTION
**Pull request by pivox**
## Summary

Part of #189  
Related to #173  
References #199, #201, #202, #203, #204

Adds a versioned, read-only Symfony API for navigating persistent trade lineage by exact identifiers, without reconstructing relationships by symbol, side, or timestamp windows.

## API surface

- `GET /api/lineage/v1/search`
  - supports `orchestration_run_id`, `correlation_run_id`, `orchestration_set_id`, `orchestration_dashboard_id`, `internal_trade_id`, `internal_position_id`, `order_intent_id`, `client_order_id`, `exchange_order_id`, `position_id`
  - requires `exchange + market_type` for `client_order_id`, `exchange_order_id`, and `position_id`
  - applies bounded pagination with `limit <= 100`
- `GET /api/lineage/v1/{internal_trade_id}`
- `GET /api/lineage/v1/{internal_trade_id}/events`

## Behavior

- Read-only only; no trading, MTF, EntryZone, risk/leverage, SL/TP, live guard, or strategy changes.
- No symbol-only fallback and no time-window matching.
- Ambiguous exact identifiers in the same venue return `409 identifier_conflict` and include `completeness_status=identifier_conflict` plus `quality_flags`.
- Every lineage item includes `completeness_status` and `quality_flags`.
- Responses are redacted by whitelist and do not expose lifecycle `extra`, `OrderIntent.rawInputs`, validation errors, provider raw payloads, or secrets.

## Completeness statuses

The read model distinguishes:

- `complete`
- `partial`
- `legacy`
- `unmatched`
- `identifier_conflict`
- `missing_order_intent`
- `missing_exchange_order_id`
- `missing_position_id`
- `missing_close_event`

## Tests

Executed locally:

- `DEFAULT_URI=http://localhost APP_ENV=test php bin/phpunit tests/Trading/Lineage/LineageContextTest.php --no-coverage` -> OK, 5 tests / 31 assertions.
- `DEFAULT_URI=http://localhost APP_ENV=test DATABASE_URL='sqlite:///:memory:' php bin/phpunit tests/Trading/Lineage tests/Trading/Controller/Api/LineageReadApiControllerTest.php --no-coverage` -> OK, 36 tests / 178 assertions.
- `DEFAULT_URI=http://localhost APP_ENV=test DATABASE_URL='postgresql://postgres:password@localhost:5436/trading_app?serverVersion=15&charset=utf8' php bin/phpunit --fail-on-skipped tests/Trading/Lineage/LineageReadIntegrationTest.php --no-coverage` -> OK, 5 tests / 19 assertions.
- `DEFAULT_URI=http://localhost APP_ENV=test DATABASE_URL='sqlite:///:memory:' vendor/bin/phpstan analyse --no-progress src/Trading/Lineage/ReadModel src/Trading/Controller/Api/LineageReadApiController.php src/Repository/TradeLineageRepository.php src/Repository/OrderIntentRepository.php src/Repository/TradeLifecycleEventRepository.php tests/Trading/Lineage/LineageReadServiceTest.php tests/Trading/Lineage/LineageReadIntegrationTest.php tests/Trading/Controller/Api/LineageReadApiControllerTest.php --memory-limit=1G` -> OK.
- `DEFAULT_URI=http://localhost APP_ENV=test DATABASE_URL='sqlite:///:memory:' php bin/console lint:container --no-debug` -> OK, with pre-existing PHP 8.4 deprecation in `EntryZoneStatsCommand`.
- `DEFAULT_URI=http://localhost APP_ENV=test php bin/console lint:yaml config` -> OK.
- `git diff --check` -> OK.

## Remaining #189 audit

This PR is `Part of #189`, not `Closes #189`. It delivers the read API requested after #199/#201/#202/#203/#204. After merge, #189 should be re-evaluated against any remaining criteria such as broader cockpit integration or additional lineage propagation/backfill work that may still be outside this read-only surface.

### CODE REVIEW

**Comment by pivox** [comment]
@codex please review this PR. Focus on correctness, regressions, architecture boundaries, missing tests, exact-identifier lineage semantics, redaction, pagination, and edge cases. Please post review threads on specific lines when useful, or a PR comment if the feedback is general.

**Comment by pivox** [comment]
@codex review

CI is green. Please review this PR again before merge, focusing on correctness, exact identifier semantics, conflict handling, pagination, and redaction.

File: trading-app/src/Repository/TradeLifecycleEventRepository.php [hunk 1]
```diff
@@ -43,6 +57,126 @@ public function findRecentBy(array $criteria, int $limit = 50, ?ExchangeContext
...
+            $or[] = 'event.orderId = :exchangeOrderId';
+            $qb->setParameter('exchangeOrderId', $exchangeOrderId);
+        }
+
+        if ($positionId !== null && $positionId !== '') {
+            $or[] = 'event.positionId = :positionId';
+            $qb->setParameter('positionId', $positionId);
+        }
+
+        $qb->andWhere($qb->expr()->orX(...$or));
```
[the remark this answers is not in the document]
Comment (pivox) [reply trading-app/src/Repository/TradeLifecycleEventRepository.php:117]: Fixed in follow-up PR #206. Lifecycle event reads now keep events with a different internal_trade_id out of the lineage detail, while still allowing legacy events with NULL internal_trade_id to match by exact venue-scoped identifiers.

File: trading-app/src/Trading/Lineage/ReadModel/LineageReadService.php [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+                items: $items,
+                total: count($items),
+                limit: $criteria->limit,
+                offset: $criteria->offset,
+            );
+        }
+
+        $items = [];
+        foreach ($lineages as $lineage) {
+            $events = $this->store->findEventsForLineage($lineage, self::EVENT_LIMIT);
```
[the remark this answers is not in the document]
Comment (pivox) [reply trading-app/src/Trading/Lineage/ReadModel/LineageReadService.php:47]: Fixed in follow-up PR #206. Completeness now checks for position_closed with a dedicated unpaginated existence query, so missing_close_event is not derived from a truncated lifecycle events page.

File: trading-app/src/Trading/Controller/Api/LineageReadApiController.php [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+    }
+
+    #[Route('/{internalTradeId}/events', name: 'events', methods: ['GET'])]
+    public function events(string $internalTradeId, Request $request): JsonResponse
+    {
+        $criteria = LineageReadCriteria::forIdentifier(
+            'internal_trade_id',
+            $internalTradeId,
+            $this->intQuery($request, 'limit', LineageReadCriteria::MAX_LIMIT),
+            $this->intQuery($request, 'offset', 0),
```
[the remark this answers is not in the document]
Comment (pivox) [reply trading-app/src/Trading/Controller/Api/LineageReadApiController.php:69]: Fixed in follow-up PR #206. The /events endpoint now performs the internal_trade_id lineage lookup with limit 1 / offset 0, then applies request limit and offset to lifecycle events.

File: trading-app/src/Trading/Lineage/ReadModel/LineageReadService.php [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+        if ($criteria->isConflictSensitive() && $total > 1) {
+            throw new LineageReadException(
+                'identifier_conflict',
+                'The identifier matches more than one lineage in the requested venue.',
+                Response::HTTP_CONFLICT,
+            );
+        }
+
+        $lineages = $this->store->find($criteria);
+        if ($lineages === []) {
```
[the remark this answers is not in the document]
Comment (pivox) [reply trading-app/src/Trading/Lineage/ReadModel/LineageReadService.php:33]: Fixed in follow-up PR #206. Search now preserves a non-zero total when the requested offset returns an empty lineage page, and only falls back to unmatched events when total is zero.
